benchmark/index.js

```diff
@@ -53,33 +51,27 @@ const shOptions = {
   functionNextLine,
 }
 
-suite
-  .add('sh-syntax', async () => {
-    await print(text, shOptions)
-  })
-  .add('mvdan-sh', () => {
-    const { syntax } = sh
-    syntax
-      .NewPrinter(
-        syntax.Indent(indent),
-        syntax.BinaryNextLine(binaryNextLine),
-        syntax.SwitchCaseIndent(switchCaseIndent),
-        syntax.SpaceRedirects(spaceRedirects),
-        syntax.KeepPadding(keepPadding),
-        syntax.Minify(minify),
-        syntax.FunctionNextLine(functionNextLine),
-      )
-      .Print(
-        syntax
-          .NewParser(syntax.KeepComments(keepComments), syntax.Variant(variant))
-          .Parse(text, filePath),
-      )
-  })
-  .on('cycle', event => {
-    console.log(String(event.target))
-  })
-  .on('complete', function () {
-    // eslint-disable-next-line @babel/no-invalid-this
-    console.log('Fastest is ' + this.filter('fastest').map('name'))
-  })
-  .run({ async: true })
+baseline('sh-syntax', () => print(text, shOptions))
```

Comment:
not awaiting `print()` causes memory leak/exhaustion to happen, this can easily be fixed by marking function as async so mitata knows that it needs to be awaited

```suggestion
baseline('sh-syntax', async () => print(text, shOptions))
```

Reply:
Thanks for the quick answer, maybe the limitation should be documented. I searched the README file, but could not find any reference about it.